RFCs/0014-TestCase-FullyQualifiedName.md

```diff
@@ -0,0 +1,48 @@
+# 0014 Guidelines for setting the FullyQualifiedName on TestCase
+
+## Summary
+This note clarifies the guidelines that test adapters should consider when setting the FullyQualifiedName on test cases before returning them from the discovery process.
+
+## Overview
+There are a broad variety of choices that current Test Adapter implementers have made when setting the FullyQualifiedName on TestCase objects created during test discovery. This note lays out some basic guidelines to make the usage of FullyQualifiedName more useful across adapters.
+
+## Usage of FullyQualifiedName
+The Test Platform uses FullyQualifiedName as a string moniker to name tests during discovery. Visual Studio may filter tests by FullyQualifiedName in order to more quickly filter and run specific tests. The guidelines below clarify the usage of FullyQualifiedName so that these optimized test execution scenarios can be realized.
+
+## Guidelines
+1. **FullyQualifiedName does NOT need to be unique across your test set.** Internally, TestCase objects are tracked by a unique ID (GUID) which can either be set by the Test Adapter implementor, or automatically generated otherwise. The FullyQualifiedName should be unique if possible, but it's not a requirement.
+1. **FullyQualifiedName should be deterministic** FullyQualifiedName should have the same value for the same test across different discovery sessions. This means that including random data, time-dependent data or environment specific data in the FullyQualifiedName should be avoided.
+1. **FullyQualifiedName is expected to be a reference to the method that implements the test** The Test Platform is expecting the FullyQualifiedName to refer to a particular test case. If there are multiple tests that are instantiated from a test method (data-driven tests for instance), then the FullyQualifiedName should distinguish these using parentheses to represent data parameters and angle brackets to represent type parameters. See below.
+
+## Recommended Form
+FullyQualifiedName is expected to be a reference to the TestCase method that implements the test (or tests). The FullyQualifiedName should be unique to a particular TestCase instance, if possible.
```

Comment:
I'm wondering when FullyQualifiedName would be used and when would its normalized form be used. Is the below the right usage?
FQN:
1. Test Explorer Group By providers(Class/Namespace).
2. Default `TestCase.ID` generation.
3. Existing Filtering.

NFQN:
1. Fast Filtering.

Reply:
NFQN will be set by the platform ? or Adapters needs to set?

Reply:
Yes, that's basically right. The main reason for creating the normalization rule is so that the TestWindow can filter tests without reproducing the FQN generation logic of each test framework. The normalization is currently implemented using the Test Window's replacement regex.

Reply:
@Navin22 my previous comment was a response to @AbhitejJohn 

NFQN does not need to be set by the platform or the adapter. I am looking to document the process we are following in the VS TestWindow.